tests/test_integration.py

```diff
@@ -1803,3 +1805,54 @@ def test_disable_cache():
     ).assert_success()
 
     assert not os.path.exists(pex_root)
+
+
+def test_unzip_mode():
+  with temporary_dir() as td:
+    pex_root = os.path.join(td, 'pex_root')
+    pex_file = os.path.join(td, 'pex_file')
+    src_dir = os.path.join(td, 'src')
+    with safe_open(os.path.join(src_dir, 'example.py'), 'w') as fp:
+      fp.write(
+        dedent(
+          """
+          import os
+          import sys
+
+          if 'quit' == sys.argv[-1]:
+              print(os.path.realpath(sys.argv[0]))
+              sys.exit(0)
```

Comment:
This part confuses me. It doesn't look like `quit` is ever the last arg in the `output{1,2}` runs, so this never gets triggered?

Reply:
The `sys.argv[:-1]` below pops the right arg off the list. If it didn't, as you note the conditional would never trigger and this would be an infinite re-exec loop.

Reply:
Ahhhh that's the part I was missing. This recursively calls itself via `sys.argv[0]`. Got it. Thanks!